pkg/common/cns-lib/volume/util.go

```diff
@@ -350,8 +366,12 @@ func ExtractFaultTypeFromErr(ctx context.Context, err error) string {
 		faultType = reflect.TypeOf(soapFault.VimFault()).String()
 		log.Infof("Extract vimfault type: +%v. SoapFault Info: +%v from err +%v", faultType, soapFault, err)
 		slice := strings.Split(faultType, ".")
-		vimFaultType := vimFaultPrefix + slice[1]
-		return vimFaultType
+		if classifyAsNonStorageFault(slice[1]) {
+			faultType = csifault.CSINonStorageFaultPrefix + slice[1]
```

Comment:
This changes the doc in line#357. Does it affect the callers?

Reply:
+1. Please also update the comment of this function to reflect the change.

Reply:
Updated method doc. No effect to the callers as they're just propagating the faulttype for metrics label.